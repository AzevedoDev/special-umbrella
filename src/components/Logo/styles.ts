import styled, { css } from 'styled-components'
import { LogoProps } from '.'

export const WrapperSvg = styled.div<LogoProps>`
  ${({ theme, isMobile }) => css`
    width: 6rem;
    height: 6rem;
    color: ${theme.colors.white};
    ${!isMobile && `margin-right: ${theme.spacings.small}`}
  `}
`
export const Wrapper = styled.div<LogoProps>`
  ${({ isMobile }) => css`
    display: flex;
    flex-direction: ${isMobile ? 'column' : 'row'};
    justify-content: center;
    align-items: center;
  `}
`
export const Text = styled.h1`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xxlarge};
    font-weight: ${theme.font.bold};
    color: ${theme.colors.white};
  `}
`
