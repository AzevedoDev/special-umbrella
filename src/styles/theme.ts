export default {
  grid: {
    container: '130rem',
    gutter: '3.2rem'
  },
  border: {
    radius: '0.4rem'
  },
  font: {
    family:
      "Montserrat, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif",
    regular: 'regular',
    normal: 500,
    bold: 700,
    sizes: {
      xsmall: '1.2rem',
      small: '1.4rem',
      medium: '1.6rem',
      large: '1.8rem',
      xlarge: '2.0rem',
      xxlarge: '2.8rem'
    }
  },
  colors: {
    mainBg: '#454545',
    white: '#FFFFFF',
    black: '#333333',
    mega: '#6BEFA3',
    quina: '#8666EF',
    lotoFacil: '#DD7AC6',
    lotoMania: '#FFAB64',
    timeMania: '#5AAD7D',
    diaSorte: '#BFAF83'
  },
  spacings: {
    xxsmall: '0.8rem',
    xsmall: '1.6rem',
    small: '2.4rem',
    medium: '3.2rem',
    large: '4.0rem',
    xlarge: '4.8rem',
    xxlarge: '5.6rem'
  },
  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50
  }
}
