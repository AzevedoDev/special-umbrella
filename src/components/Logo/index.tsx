import * as S from './styles'

export type LogoProps = {
  color?: 'white' | 'black'
  size?: 'normal' | 'small'
  isMobile?: boolean
  title?: string
}

const Logo = ({
  color = 'white',
  size = 'normal',
  isMobile = false,
  title = 'Sena'
}: LogoProps) => (
  <S.Wrapper color={color} isMobile={isMobile} data-testid="logo-component">
    <S.WrapperSvg color={color} size={size} isMobile={isMobile}>
      <svg
        fill="none"
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 65 64"
        aria-label={title}
        role="img"
      >
        <path
          d="M21.517 63.154a9.26 9.26 0 009.241-9.241V33.271H9.206a9.243 9.243 0 103.5 17.84 9.234 9.234 0 008.811 12.043zM42.623.754a9.26 9.26 0 00-9.24 9.241v20.64h21.549a9.241 9.241 0 10-3.5-17.84A9.235 9.235 0 0042.623.754zM21.517.754a9.26 9.26 0 019.241 9.241v20.641H9.206a9.24 9.24 0 113.5-17.84A9.236 9.236 0 0121.517.754zm21.106 62.4a9.259 9.259 0 01-9.24-9.242V33.271h21.549a9.242 9.242 0 11-3.5 17.84 9.236 9.236 0 01-8.809 12.043z"
          fill="currentColor"
        />
      </svg>
    </S.WrapperSvg>
    <S.Text>{title}</S.Text>
  </S.Wrapper>
)

export default Logo
