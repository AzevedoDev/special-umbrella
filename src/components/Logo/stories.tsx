import { Story, Meta } from '@storybook/react/types-6-0'
import Logo, { LogoProps } from '.'

export default {
  title: 'Logo',
  component: Logo,
  parameters: {
    backgrounds: {
      default: 'mega',
      values: [
        {
          name: 'mega',
          value: '#6BEFA3'
        },
        {
          name: 'quina',
          value: '#8666EF'
        },
        {
          name: 'lotoFacil',
          value: '#DD7AC6'
        },
        {
          name: 'lotoMania',
          value: '#FFAB64'
        },
        {
          name: 'timeMania',
          value: '#5AAD7D'
        },
        {
          name: 'diaSorte',
          value: '#BFAF83'
        }
      ]
    }
  }
} as Meta

export const Default: Story<LogoProps> = (args) => <Logo {...args} />
