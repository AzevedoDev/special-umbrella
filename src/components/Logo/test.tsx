import { screen } from '@testing-library/react'
import theme from 'styles/theme'
import { renderWithTheme } from 'utils/tests/helpers'

import Logo from '.'

describe('<Logo />', () => {
  it('should render and match with snapshot', () => {
    renderWithTheme(<Logo />)
    expect(screen.getByLabelText(/Sena/i)).toMatchSnapshot()
  })

  it('should render a white logo by default', () => {
    renderWithTheme(<Logo />)
    expect(screen.getByLabelText(/Sena/i).parentElement).toHaveStyle({
      color: theme.colors.white
    })
  })

  it('should render a normal size logo', () => {
    renderWithTheme(<Logo />)
    expect(screen.getByLabelText(/Sena/i).parentElement).toHaveStyle({
      width: '6rem'
    })
  })

  it('should render a when a isMobile Props is true', () => {
    renderWithTheme(<Logo isMobile />)
    expect(screen.getByTestId('logo-component')).toHaveStyle({
      flexDirection: 'column'
    })
  })
})
