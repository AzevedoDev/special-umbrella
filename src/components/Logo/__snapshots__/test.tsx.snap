// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render and match with snapshot 1`] = `
<svg
  aria-label="Sena"
  fill="none"
  role="img"
  viewBox="0 0 65 64"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M21.517 63.154a9.26 9.26 0 009.241-9.241V33.271H9.206a9.243 9.243 0 103.5 17.84 9.234 9.234 0 008.811 12.043zM42.623.754a9.26 9.26 0 00-9.24 9.241v20.64h21.549a9.241 9.241 0 10-3.5-17.84A9.235 9.235 0 0042.623.754zM21.517.754a9.26 9.26 0 019.241 9.241v20.641H9.206a9.24 9.24 0 113.5-17.84A9.236 9.236 0 0121.517.754zm21.106 62.4a9.259 9.259 0 01-9.24-9.242V33.271h21.549a9.242 9.242 0 11-3.5 17.84 9.236 9.236 0 01-8.809 12.043z"
    fill="currentColor"
  />
</svg>
`
